Updated documentation for retries and circuit breakers (#1507)
datawire/ambassador: docs/reference/circuit-breakers.md

(Default: `1024`) Specifies the maximum number of connections that Ambassador will make to the services. In practice, this is more applicable to HTTP/1.1 than HTTP/2.

### `max_pending_requests`
(Default: `1024`) Specifies the maximum number of requests that will be queued while waiting for a connection. In practice, this is more applicable to HTTP/1.1 than HTTP/2.

### `max_requests`
(Default: `1024`) Specifies the maximum number of parallel outstanding requests to hosts. In practice, this is more applicable to HTTP/2 than HTTP/1.1.

### `max_retries`
(Default: `3`) Specifies the maximum number of parallel retries allowed to hosts.

## Examples

Circuit breakers defined on a single mapping:

```yaml
apiVersion: ambassador/v1
kind:  Mapping
name:  qotm_mapping
prefix: /qotm/
service: qotm
circuit_breakers:
- max_connections: 2048
  max_pending_requests: 2048
```

A global circuit breaker:

```yaml
apiVersion: ambassador/v0
kind:  Module
name:  ambassador
config:
  circuit_breakers:
  - max_connections: 2048
    max_pending_requests: 2048
---
apiVersion: ambassador/v1
kind:  Mapping
name:  qotm_mapping
prefix: /qotm/
service: qotm
```

## Circuit breakers and automatic retries

> Circuit breakers are designed to be used in conjunction with [automatic retries](reference/retries). Here are some examples:

Reviewer: ```suggestion
Circuit breakers are best used in conjunction with [automatic retries](reference/retries). Here are some examples:
```

Reviewer: Basically, I want to indicate that it's a best practice instead of that circuit breakers are incomplete without retries.
